File: ps-backend/src/main/java/org/commonground/ps/backendapi/model/PageButton.java
package org.commonground.ps.backendapi.model;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class PageButton {
  private Long id;
  @NotNull(message = "Waarde is verplicht")
  @Size(min = 1, max = 100, message = "Waarde is minimaal 1 en maximaal 100 tekens")
  private String name;
  @Size(max = 512, message = "Waarde is minimaal 1 en maximaal 512 tekens")
  private String route;
  private ActionType action;
  @NotNull(message = "Waarde is verplicht")
  @Size(min = 1, max = 100, message = "Waarde is minimaal 1 en maximaal 100 tekens")
  private String type;
  @Valid
  private List<Role> roles;
  @Valid
  private List<PageButtonCondition> conditions;

  @JsonIgnore
  private PageOverviewTemplate pageOverview;
}

File: ps-frontend/src/model/page-button.ts
import { ActionType } from "./action-type";
import { ButtonTypes } from "./intefaces";
import { PageButtonCondition } from "./page-button-condition";
import { Role } from "./role";

export class PageButton {
    public id: number;
    public name: string;
    public route: string;
    public action: ActionType;
    public type: ButtonTypes;
    public roles?: Role[];
    public conditions?: PageButtonCondition[];
  }
  